apps/dashboard/src/components/project-navigation/index.js

```diff
@@ -21,24 +21,29 @@ const Tab = {
 };
 
 const ProjectNavigation = ( { activeTab, disableTitleEditor, projectId } ) => {
-	const { saveAndUpdateProject } = useDispatch( STORE_NAME );
+	const { setEditorTitle } = useDispatch( STORE_NAME );
 
-	const project = useSelect(
-		( select ) => select( STORE_NAME ).getProject( projectId ),
+	const projectTitle = useSelect(
+		( select ) => select( STORE_NAME ).getEditorTitle(),
 		[ projectId ]
 	);
 
-	const updateTitle = ( title ) =>
-		saveAndUpdateProject( projectId, {
-			name: title,
-			title,
-		} );
+	const { setEditorContentChanged } = useDispatch( STORE_NAME );
+
+	const updateTitle = ( title ) => {
+		if ( title !== projectTitle ) {
+			setEditorTitle( title );
+			setEditorContentChanged();
+		}
+
+		return true;
+	};
 
 	return (
 		<div className="project-navigation">
 			<EditablePageHeader
 				onChange={ updateTitle }
-				text={ project?.title || __( 'Untitled Project', 'dashboard' ) }
+				text={ projectTitle || __( 'Untitled Project', 'dashboard' ) }
```

Comment:
Note that `projectTitle` can be empty if you're loading an existing project into a 'new' editor.  
So this needs to be `projectTitle || project?.title || __( 'Untitled Project', 'dashboard' )`.

Reply:
I tried it with an existing project and it worked fine. Not sure what you mean by *'new' editor*